js/geo/transform.js

```diff
@@ -375,7 +394,8 @@ Transform.prototype = {
                 y2 !== undefined ? y2 : this.y));
         }
 
-        this._constraining = false;
+        this._unmodified = this._constraining.unmodified;
+        this._constraining = undefined;
```

Comment:
Elsewhere `_constrain()` calls the `center` and `zoom` setters, and I want this specific case of the map being repositioned by `_constrain()` to not affect the unmodified flag, whereas all other calls to setters should affect it.

Maybe renaming `unmodified` to `modified` and swapping all the logic might make this easier to reason about too.    So today I will add some comments and rename that variable.

`_constrain()` itself is a kind of problematic in that it introduces transform side effects and I don't like that this private function is being called by `Map#resize`.  Maybe later we can think of a better way to refactor `Transform` into smaller responsibility classes.  I don't like having a big class that is responsible for the map state, the WGS84 projection code, and the scene view (camera).